File: Lab5/C#/lab5/Section.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace lab5
{
    class Section : Line
    {
        public Section() : base(new Tochka(), new Tochka()) { }
        public Section(Tochka s, Tochka e) : base(s, e) { }
        
        public Section Increase2()
        { 
            Tochka tempStart = new Tochka( start * 2);
            Tochka tempEnd = new Tochka(end * 2);
            return new Section(tempStart, tempEnd);
        }
        public Tochka StartCoord => start;
        public Tochka EndCoord => end;

    }
}

File: Lab5/C++/lab5/Section.cpp
#include "Section.h"

Section::Section() : Line() { };
Section::Section(Tochka s, Tochka e) : Line(s, e) { };

Section Section:: Increase2()
{
	Section res;
	res.start = res.start * 2;
	res.end = res.end * 2;
	return res;
}

Tochka Section::StartCoord() { return start; }
Tochka Section::EndCoord() { return end; }